strict-stm/strict-stm.cabal

```diff
@@ -0,0 +1,66 @@
+name:                strict-stm
+version:             0.1.0.0
+synopsis:            Type classes for concurrency with STM, ST and timing
+-- description:
+license:             Apache-2.0
+license-files:
+  LICENSE
+  NOTICE
+copyright:           2019-2021 Input Output (Hong Kong) Ltd.
+author:              Alexander Vieth, Marcin Szamotulski, Duncan Coutts
+maintainer:
+category:            Control
+build-type:          Simple
+cabal-version:       >=1.10
+
+source-repository head
+  type:     git
+  location: https://github.com/input-output-hk/ouroboros-network
+  subdir:   io-classes
+
+flag checktvarinvariant
+  Description: Enable runtime invariant checks on StrictT(M)Var
+  Manual: True
+  Default: False
+
+flag asserts
+  description: Enable assertions
+  manual:      False
+  default:     False
+
+library
+  hs-source-dirs:      src
+
+  -- At this experiment/prototype stage everything is exposed.
+  -- This has to be tidied up once the design becomes clear.
+  exposed-modules:     Control.Monad.Class.MonadSTM.Strict
+  default-language:    Haskell2010
+  build-depends:       base  >=4.9 && <4.15,
+                       stm   >=2.5 && <2.6,
+                       io-classes
+  ghc-options:         -Wall
+                       -Wno-unticked-promoted-constructors
+                       -Wcompat
+                       -Wincomplete-uni-patterns
+                       -Wincomplete-record-updates
+                       -Wpartial-fields
+                       -Widentities
+
+  if flag(asserts)
+    ghc-options: -fno-ignore-asserts
+
+  if flag(checktvarinvariant)
+    cpp-options: -DCHECK_TVAR_INVARIANT
+
+-- test-suite test
```

Comment:
Do we want to remove this?

Reply:
I don't think so, don't you use the invariant somewhere in `ouroboros-cosnensus`?  Even if not, I think it's still useful.

Reply:
Sorry, I meant the commented code :)

Reply:
Oh, yes it doesn't make sense here.